lib/checkWin.js

```diff
@@ -0,0 +1,71 @@
+const logGrid = require("./logGrid");
+const playAgain = require("./playAgain");
+const { setWinner } = require("./store");
+
+const mapGridValues = (store) => {
+  let gridArray = [
+    [1, 2, 3],
+    [4, 5, 6],
+    [7, 8, 9],
+    [1, 4, 7],
+    [2, 5, 8],
+    [3, 6, 9],
+    [1, 5, 9],
+    [3, 5, 7],
+  ];
+  return gridArray.map((array) => {
+    let result = array
+      .map((ref) => {
+        return store.grid[ref];
+      })
+      .join("");
+    return result;
+  });
+};
+
+const checkMappedGridForWin = (array) => {
+  let winner = false;
+  array.map((result) => {
+    if (result === "XXX" || result === "OOO") {
+      winner = result.split("")[0];
+    }
+  });
+  return winner;
+};
+
+const checkForDraw = (store) => {
+  if (store.choices.length === 9 && store.winner === false) {
+    console.log("You drew!");
+    return true;
+  } else {
+    return false;
+  }
+};
+
+const checkForWin = (store) => {
+  let winner = checkMappedGridForWin(mapGridValues(store));
+  if (winner === store.userTeam) {
+    logGrid(store);
+    console.log("You won!");
+    return playAgain();
+  } else if (winner === store.cpuTeam) {
+    logGrid(store);
+    console.log("CPU won!");
+    return playAgain();
+  }
+};
+
+const checkForGameOver = (store) => {
+  checkForWin(store);
+  if (checkForDraw(store)) {
+    return playAgain();
```

Comment:
... actually, the top-level loop exits if the game is over, so that won't cause the behaviour we're seeing

The behaviour will happen if there is a win when the last piece is played

* checkForWin() triggers a new game because it detects a win
* checkForDraw() triggers a new game because it detects a draw (9 pieces are placed)

If you wanted to solve this bug, you could make checkForDraw guard against boards with a winner and do nothing in that case

In the long term, I would suggest using "structured concurrency" principle, and *always* await things in a tree. This will stop you from having this class of problem entirely, but might not be easy to do with your current recursive approach. I'm happy to sketch out the call tree with you in miro, if you want.